zerver/models.py

```diff
@@ -2897,24 +2894,13 @@ def fetch_streams_by_name(stream_names: List[str]) -> QuerySet[Stream]:
         where_clause = (
             "upper(zerver_stream.name::text) IN (SELECT upper(name) FROM unnest(%s) AS name)"
         )
-        return (
-            get_active_streams(realm)
-            .select_related()
-            .extra(where=[where_clause], params=(list(stream_names),))
-        )
-
-    def stream_name_to_cache_key(stream_name: str) -> str:
-        return get_stream_cache_key(stream_name, realm.id)
+        return get_active_streams(realm).extra(where=[where_clause], params=(list(stream_names),))
 
-    def stream_to_lower_name(stream: Stream) -> str:
-        return stream.name.lower()
-
-    return bulk_cached_fetch(
-        stream_name_to_cache_key,
-        fetch_streams_by_name,
-        [stream_name.lower() for stream_name in stream_names],
-        id_fetcher=stream_to_lower_name,
-    )
+    if not stream_names:
+        return {}
+    lower_stream_names = [stream_name.lower() for stream_name in stream_names]
```

Comment:
I think this is unnecessary, `where_clause` adjusts case anyway?

Reply:
Yeah, good catch.  Running tests on that change.